tasmota/xdrv_10_rules.ino

```diff
@@ -471,6 +472,15 @@ bool RulesRuleMatch(uint8_t rule_set, String &event, String &rule, bool stop_all
       rule_param = String(SunMinutes(1));
     }
 #endif  // USE_TIMERS and USE_SUNRISE
+#if defined(USE_TIMERS)
+    for (uint32_t i = 0; i < MAX_TIMERS; i++) {
+      snprintf_P(stemp, sizeof(stemp), PSTR("%%TIMER%d%%"), i +1);
+      if (rule_param.startsWith(stemp)) {
+        rule_param = String(Settings->timer[i].time);
+        break;
+      }
+    }
```

Comment:
Please optimize this
Comparing to 16 strings is non sense
Search first for `%timer` and only if found, look for the number and the ending

Reply:
Thank you
I optimize.

Reply:
optimize mohno like this: 
    if (rule_param.startsWith(F("%TIMER"))) {
      int index = rule_param.substring(5).toInt();
      if (index > 0 && index <= MAX_TIMERS) {
         snprintf_P(stemp, sizeof(stemp), PSTR("%%TIMER%d%%"), index);
         if (rule_param.startsWith(stemp))
           rule_param = String(Settings->timer[index-1].time);
      }
    }
this option will run faster and take more code.